ci/jenkins/test-rancher.sh

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+DEFAULT_WORKDIR=$HOME
+WORKDIR=$DEFAULT_WORKDIR
+KUBECONFIG_PATH="$HOME/.kube/config"
+TESTCASE=""
+TEST_FAILURE=false
+MODE="report"
+DOCKER_REGISTRY=$(head -n1 "${WORKSPACE}/ci/docker-registry")
+# Dynamic cluster creation for Rancher cluster, using vSphere as IaaS is not supported for now.
+TESTBED_TYPE="static"
+GO_VERSION=$(head -n1 "${WORKSPACE}/build/images/deps/go-version")
+IMAGE_PULL_POLICY="Always"
+
+CONFORMANCE_SKIP="\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|\[sig-cli\]|\[sig-storage\]|\[sig-auth\]|\[sig-api-machinery\]|\[sig-apps\]|\[sig-node\]"
+NETWORKPOLICY_SKIP="should allow egress access to server in CIDR block|should enforce except clause while egress access to server in CIDR block"
+
+# In Rancher clusters the control-plane node role is "controlplane,etcd".
+CONTROL_PLANE_NODE_ROLE="controlplane,etcd"
+CLUSTER_NAME=""
+
+CLEAN_STALE_IMAGES="docker system prune --force --all --filter until=48h"
+
+_usage="Usage: $0 [--kubeconfig <KubeconfigSavePath>] [--workdir <HomePath>]
+                  [--testcase <e2e|conformance|networkpolicy]
+
+Run K8s e2e community tests (Conformance & Network Policy) or Antrea e2e tests on a remote (Jenkins) Rancher cluster.
+
+        --cluster-name           Name of the kubernetes cluster.
+        --workdir                Home path for Go, vSphere information and antrea_logs during cluster setup. Default is $WORKDIR.
+        --testcase               Conformance and Network Policy or Antrea e2e testcases on a Rancher cluster.
+        --registry               The docker registry to use instead of dockerhub.
+        --testbed-type           The testbed type to run tests. It can be static, or dynamic, dynamic is not supported for now."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --cluster-name)
+    CLUSTER_NAME="$2"
+    shift 2
+    ;;
+    --workdir)
+    WORKDIR="$2"
+    shift 2
+    ;;
+    --testcase)
+    TESTCASE="$2"
+    shift 2
+    ;;
+    --registry)
+    DOCKER_REGISTRY="$2"
+    shift 2
+    ;;
+    --testbed-type)
+    TESTBED_TYPE="$2"
```

Comment:
ditto. At least add comments to explain why this parameter is not used in the script.